File: app/src/main/java/com/google/android/apps/gsa/nowoverlayservice/ConfigurationOverlayController.java
package com.google.android.apps.gsa.nowoverlayservice;

import android.app.Service;
import android.content.Context;
import android.content.res.Configuration;

import androidx.annotation.Nullable;

import com.google.android.libraries.gsa.d.a.OverlayController;
import com.google.android.libraries.gsa.d.a.OverlaysController;
import com.google.android.libraries.gsa.d.a.v;

import ua.itaysonlab.homefeeder.overlay.Overlay;

public final class ConfigurationOverlayController extends OverlaysController {

    private final Context mContext;

    public ConfigurationOverlayController(Service service) {
        super(service);
        this.mContext = service;
    }

    //Todo: was protected
    public final int Hx() {
        return 24;
    }

    public final OverlayController createController(@Nullable Configuration configuration, int i, int i2) {
        Context context = this.mContext;
        if (configuration != null) {
            context = context.createConfigurationContext(configuration);
        }
        return new Overlay(context, i, i2);
    }

    //Todo: was protected, and return modified
    public final v HA() {
        return new v();
    }
}

File: app/src/main/java/com/google/android/apps/gsa/nowoverlayservice/ConfigurationOverlayController.kt
package com.google.android.apps.gsa.nowoverlayservice

import android.app.Service
import android.content.Context
import android.content.res.Configuration

import com.google.android.libraries.gsa.d.a.OverlayController
import com.google.android.libraries.gsa.d.a.OverlaysController
import com.google.android.libraries.gsa.d.a.v

import ua.itaysonlab.homefeeder.overlay.OverlayKt

class ConfigurationOverlayController(private val service: Service) : OverlaysController(service) {
    override fun Hx() = 24

    override fun createController(
        configuration: Configuration?,
        i: Int,
        i2: Int
    ): OverlayController = OverlayKt(if (configuration != null) service.createConfigurationContext(configuration) else service)

    override fun HA() = v()
}
